mirai-core-qqandroid/src/commonMain/kotlin/net/mamoe/mirai/qqandroid/utils/AtomicResizeCacheList.kt

```diff
@@ -73,9 +73,7 @@ internal class AtomicResizeCacheList<E>(private val retention: Long) {
      * No concurrency guaranteed on same [element]
      */
     private fun removeDuplication(element: E): Boolean {
-        val duplicate = list.firstOrNull { it.time.value != 0L && it.element == element } ?: return false
-        duplicate.time.value = 0
-        return true
+        return list.any { it.time.value != 0L && it.element == element }
     }
```

Comment:
Setting `time = 0` is to indicate that such element is no longer needed, and is neccessary here.